test/extensions/http/stateful_session/cookie/cookie_test.cc

```diff
@@ -52,7 +52,6 @@ TEST(CookieBasedSessionStateFactoryTest, SessionStateTest) {
       if (use_proto) {
         envoy::Cookie cookie;
         cookie.set_address("1.2.3.4:80");
-        cookie.set_expires(1000);
```

Comment:
Why is this removed from the test?

If the cookie `expires` field is the same as the current time (i.e., `now == expiry_time` in the code above), what should be the returned value? The test here seems to send a set-cookie with TTL of 0 (which AFAICT is infinite), no? I'm wondering if this is the right behavior.

Reply:
Right, this is testing with TTL 0. `cookie` here is the expected value. The correct expected value should have expiry infinite (or 0 due to the encoding). The previous commit had it at `1000` which means it immediately expired once issued.

Reply:
I think that this test was designed to test the situation when a request without a cookie arrives to stateful filter and the filter needs to generate a cookie in response. This resembles initial request when an initial cookie is generated. The generated cookie's expiry time should be `time_now + configured ttl`. In this test configured ttl was previously zero and monotonic time is 1000, so the test checked if generated cookie's expiry time is 1000. I think we should keep this test but add EXPECT_CALL to return config's ttl which is not zero to verify that proper expiry time is generated. Additionally another test should be added to verify that when ttl is zero, the proper cookie is generated in similar, initial situation.

Reply:
TTL 0 has a different meaning for expiry - it must mean the generated cookie has no expiry at all. Checking for `time_now+configured ttl` when TTL is not 0 is done in the next test case below.

Reply:
I see, thanks for clarifying. Please add a comment to the test explaining the intent here.

Reply:
No problem, added a comment.